components/home/index.js

```diff
@@ -33,47 +40,86 @@ class Home extends Component {
         longitudeDelta: 0.05
       }
     });
-  }
+  };
 
-  // Function : change the visiblity of the switchCampuses component
-  // parameter: boolean to set the visibility (false: unvisible)
   changeVisibilityTo = (visibility) => {
     this.setState({ isVisible: visibility });
+  };
+
+  changeVisibilityToSearch = (visibility) => {
+    // eslint-disable-next-line react/no-unused-state
+    this.setState({ isSearchVisible: visibility });
+  };
+
+  changeVisibilityToGo = (visibility) => {
+    this.setState({ isGoVisible: visibility });
+  }
+
+  changeVisibilityToSwitchCampus = (visibility) => {
+    this.setState({ isSwitchAvailableIndestination: visibility });
   }
 
-  // Function: Updates coordinates state to draw polyline
-  // Parameter: object with latitudes and longitudes
+
   updateCoordinates = (newCoordinates) => {
     this.setState({
       coordinates: newCoordinates
     });
-  }
+  };
 
   getPolylinePoint = (data) => {
     this.setState({
       encryptedLine: data
     });
+  };
+
+  getRegionFromAddresses=(region) => {
+    this.updateRegion(region);
+  };
+
+  getCoordinatesFromAddresses=(coordinates) => {
+    this.updateCoordinates(coordinates);
   }
 
   render() {
     return (
       <View style={styles.container}>
+
         <TheMap
           updatedRegion={this.state.region}
           updatedCoordinates={this.state.coordinates}
           encryptedLine={this.state.encryptedLine}
         />
+        {!this.state.isGoVisible && (
         <SearchBar
           navigation={this.props.navigation}
           updateRegion={this.updateRegion}
           changeVisibilityTo={this.changeVisibilityTo}
+          changeVisibilityToSearch={this.changeVisibilityToSearch}
+        />
+        )}
+        <SwitchCampuses
+          updateRegion={this.updateRegion}
+          visiblityState={this.state.isVisible}
+          isSwitchAvailableIndestination={this.state.isSwitchAvailableIndestination}
         />
-        <SwitchCampuses updateRegion={this.updateRegion} visiblityState={this.state.isVisible} />
         <WithinBuilding />
         <Shuttle
           coordinateCallback={this.updateCoordinates}
           getPolylinePoint={this.getPolylinePoint}
         />
+        <ToCircle
```

Comment:
Can this be renamed to something that represents the component such as SetCourse